app/services/search/query_builders/user.rb

```diff
@@ -21,6 +30,19 @@ def initialize(params)
       def build_queries
         @body[:query] = { bool: {} }
         @body[:query][:bool][:must] = query_conditions if query_keys_present?
+        @body[:query][:bool][:must_not] = excluded_term_keys if excluded_term_keys_present?
+      end
+
+      def excluded_term_keys_present?
+        self.class::EXCLUDED_TERM_KEYS.detect { |key, _| @params[key].present? }
+      end
+
+      def excluded_term_keys
+        EXCLUDED_TERM_KEYS.map do |term_key, search_key|
+          next unless @params.key? term_key
+
+          { terms: { search_key => Array.wrap(@params[term_key]) } }
```

Comment:
It's more likely a result of me being burnt out today or not being as deeply familiar with Elasticsearch's naming...but this was confusing. I had to read it a few times to understand why it's done this way. I couldn't think of a better way to do it or a better way to name it though 🙈

It gets tricky because in the future we may want to filter by roles, so you have to come up with some sort of key that's different to differentiate which logic to apply. I thought maybe including an action with the key something like `action: include/exclude`. But that causes rippling changes for a very minimal "readability gain".

Reply:
Makes sense - thanks! I appreciate the follow-up to my otherwise pointless comment!

Reply:
LOL They are never pointless! It keeps refactoring at the forefront of my mind so we never get too far down the road that it turns into a huge refactor. I really think our refactoring pace, do it after 2 or 3 repeats, has been great and leads to small nice refactor PRs rather than huge ones. +1 for our team 💁